sympy/solvers/ode/single.py

```diff
@@ -945,7 +945,7 @@ class RiccatiSpecial(SinglePatternODESolver):
     >>> f = Function('f')
     >>> y = f(x)
     >>> genform = a*y.diff(x) - (b*y**2 + c*y/x + d/x**2)
-    >>> sol = dsolve(genform, y)
+    >>> sol = dsolve(genform, y, hint="Riccati_special_minus2")
```

Comment:
Is the `Ricatti_special_minus` solver still needed?

Reply:
Yes, the `1st_rational_riccati` solver doesn't work for symbolic coefficients

Reply:
That needs to be explained in comments/docstrings etc. If the coefficients need to be rational then that needs to be checked for.

Reply:
I will mention it in the docstring. We are already checking if the functions are rational, so I think that would cover it.

Reply:
I don't see anything that checks the coefficients of the rational functions. Note that `is_rational_function` does not do that:
```python
In [4]: a = z*x**2 + pi*exp(1) - 3.1

In [5]: a
Out[5]: 
 2              
x ⋅z - 3.1 + ℯ⋅π

In [6]: a.is_rational_function(x)
Out[6]: True
```